services/src/main/java/com/linkedin/kmf/consumer/KMBaseConsumer.java

```diff
@@ -14,6 +14,12 @@
  */
 public interface KMBaseConsumer {
 
+  /**
```

Comment:
Note that restriction is inconvenient for developers who want to plugin their consumer implementation in the Kafka Monitor, because their existing API may only provide a blocking API for receiving messages.

Maybe we don't necessarily need this restriction. How about we simply mark the thread as daemon and don't wait for consumer thread to shutdown when `ConsumeService.stop()` is called?